src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/Extensions/DiagnosticExtensions.cs

```diff
@@ -3,30 +3,28 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Collections.Generic;
 using System.Reflection;
-using System.Text;
 
 namespace Microsoft.CodeAnalysis.Testing.Extensions
 {
     internal static class DiagnosticExtensions
     {
-        private static readonly MethodInfo? IsSuppressedGetMethod = typeof(Diagnostic).GetProperty("IsSuppressed")?.GetMethod;
+        private static readonly Func<Diagnostic, bool> s_isSuppressed;
```

Comment:
We could just add the `CreateDelegate` to the field initializer, instead of introducing a static constructor, but I think that's just personal flavor.
```c#
        private static readonly Func<Diagnostic, bool> s_isSuppressed = (Func<Diagnostic, bool>?)typeof(Diagnostic).GetProperty(nameof(IsSuppressed), typeof(bool))
            ?.GetMethod?.CreateDelegate(typeof(Func<Diagnostic, bool>), target: null)
        ?? (_ => false);
```

Reply:
I find the current form a bit more readable. 👍